src/core/engine/processors/kitchen_simulator.py

```diff
@@ -1,13 +1,124 @@
 from abc import ABCMeta
 
 from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.lib.entities.order_status_history import OrderStatusHistory
+from src.constants.order_status import OrderStatus
+from datetime import datetime
+from src.core.ioc import get_ioc_instance
 
 
 class KitchenSimulator(AbstractProcessor, metaclass=ABCMeta):
     def __init__(self, app_processor_config, app_context=None):
         super().__init__(
             app_processor_config=app_processor_config, app_context=app_context
         )
+        self.instance_ioc = get_ioc_instance()
+        self.order_status_history_controller = self.instance_ioc.get_instance(
+            "order_status_history_controller"
+        )
 
     def process(self, delta_time):
         print("Waiting orders")
+
+    def create_new_order_status_history(
+        self, order=None, from_status=None, from_time=None, to_status=None, to_time=None
+    ):
+        new_order_status_history = OrderStatusHistory()
+        new_order_status_history.order = order
+        new_order_status_history.from_status = from_status
+        new_order_status_history.from_time = from_time
+        new_order_status_history.to_status = to_status
+        new_order_status_history.to_time = to_time
+        return new_order_status_history
+
```

Comment:
Could you handle different statses using differen processes?

Example:

process_new_orders()
process_placed_orders()
cleaned_ordes_queue()